pkg/csi_driver/controller.go

```diff
@@ -101,6 +104,28 @@ func (s *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolu
 			return nil, status.Error(codes.AlreadyExists, err.Error())
 		}
 	} else {
+		// If we are creating a new instance, we need to evaluate the reservedIPRange range if reserved-ipv4-cidr is provided
+		// If the param was not provided, we default reservedIPRange to "" and cloud provider takes care of the allocation
+		reservedIPV4CIDR, ok := req.GetParameters()[paramReservedIPV4CIDR]
+		if ok {
```

Comment:
nit: if only needed inside `if` scope you can do:
`if reservedIPV4CIDR, ok := req.GetParameters()[paramReservedIPV4CIDR]; ok {`